app/templates/events/event_list.html

```diff
@@ -98,11 +97,9 @@ <h1 class="text-center mt-5">Events List for {{selectedTerm.description}}</h1>
                 </button>
               </td>
             {% else %}
-              {% if (event.isPast) %}
-                <td><button type="button" class="btn btn-primary" disabled>RSVP</button></td>
-              {% elif event.id in rsvpedEventsID %}
+              {% if event.id in rsvpedEventsID %}
                 <td><button type="button" class="btn btn-danger" onclick="removeRsvpForEvent({{event.id}})">Remove RSVP</button></td>
-              {% else %}
+              {% elif (not event.isPast) and (event.isRsvpRequired) %}
                 <td><button type="button" class="btn btn-primary" onclick="rsvpForEvent({{event.id}})">RSVP</button></td>
```

Comment:
There always needs to be a cell, even if no button shows up in it, so the `<td>` and `</td>` should be outside of the conditionals